iota-execution/latest/iota-adapter/src/execution_engine.rs

```diff
@@ -836,35 +844,56 @@ mod checked {
         Ok(builder.finish())
     }
 
-    /// Advances the epoch by constructing a `ProgrammableTransaction` with
-    /// `construct_advance_epoch_pt` and executing it.
-    /// If the transaction fails, it switches to safe mode and retries the
-    /// epoch advancement in a more controlled environment. The function also
+    pub fn construct_advance_epoch_pt_v1(
+        builder: ProgrammableTransactionBuilder,
+        params: &AdvanceEpochParams,
+    ) -> Result<ProgrammableTransaction, ExecutionError> {
+        let call_arg_vec = vec![
+            CallArg::IOTA_SYSTEM_MUT,
```

Comment:
I would either add a comment here, that `construct_advance_epoch_pt_impl` adds more parameters here in the front, so that you directly see that this matches the "advance epoch" function, or we move the code from `construct_advance_epoch_pt_impl` and duplicate it to `_v1` and `_v2`, so that it is clearly visible?

Reply:
But yeah, not a blocking issue from my side, PR approved.

Reply:
Comments added [here](https://github.com/iotaledger/iota/commit/4adae113696317303019123ebf0a9e1b0c6ace9b)